horovod/common/operations.cc

```diff
@@ -1101,32 +1100,30 @@ void PerformOperation(TensorTable& tensor_table, MPIResponse response) {
           // a buffer is not safe since the tensor can be arbitrarily large.
           host_buffer = malloc(total_buffer_len);
 
+          // Synchronize.
+          WAIT_FOR_EVENTS(entries, timeline, event_queue)
+
+          ACTIVITY_START_ALL(entries, timeline, MEMCPY_IN_HOST_BUFFER)
           CUDA_CHECK(entries, "cudaMemcpyAsync",
                      cudaMemcpyAsync(host_buffer, buffer_data_at_rank_offset,
                                      total_buffer_len, cudaMemcpyDeviceToHost,
                                      stream))
-          // This event must be recorded for the subsequent synchronize.
-          RECORD_EVENT(entries, event_queue, MEMCPY_IN_HOST_BUFFER, stream)
-
-          // Synchronize.
-          WAIT_FOR_EVENTS(entries, timeline, event_queue)
+          ACTIVITY_END_ALL(entries, timeline)
 
           ACTIVITY_START_ALL(entries, timeline, MPI_ALLREDUCE)
```

Comment:
Let's add a comment that according to https://docs.nvidia.com/cuda/cuda-runtime-api/api-sync-behavior.html#api-sync-behavior__memcpy-async the previous `cudaMemcpyAsync` is synchronous w.r.t. the host.